pulser-core/pulser/register/weight_maps.py

```diff
@@ -159,7 +166,8 @@ def _to_abstract_repr(self) -> dict[str, Any]:
             traps=[
                 {"weight": weight, "x": x, "y": y}
                 for weight, (x, y) in zip(
-                    self.sorted_weights, self.sorted_coords
+                    self.sorted_weights,
+                    self.sorted_coords.as_array(detach=True),
```

Comment:
I think these sort of changes shouldn't be necessary. I guess if we `detach=True` in `AbstractArray._to_abstract_repr()` it will be fine

Reply:
Correct, the change in this particular place is not necessary - it's a remnant of previous solution. Now `mypy` doesn't complain about it.